src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1267,9 +1267,11 @@ abstract class RefChecks extends Transform {
     // warnings after the first, but I think it'd be better if we didn't have to
     // arbitrarily choose one as more important than the other.
     private def checkUndesiredProperties(sym: Symbol, pos: Position): Unit = {
-      // If symbol is deprecated, and the point of reference is not enclosed
-      // in either a deprecated member or a scala bridge method, issue a warning.
-      if (sym.isDeprecated && !currentOwner.ownerChain.exists(x => x.isDeprecated))
+      // Issue a warning if symbol is deprecated, unless the point of reference is enclosed by a deprecated member,
+      // or an element that is synthetic or has a deprecated companion.
+      if (sym.isDeprecated &&
+          (!currentOwner.isSynthetic || currentOwner.isAnonymousFunction || currentOwner.isParameter) &&
+          !currentOwner.ownersIterator.exists(s => s.isDeprecated || s.companionClass.isDeprecated))
```

Comment:
The condition for `isSynthetic` is unfortunately tweaky. Is there a better way to ask, Did the user introduce this symbol? I might have guessed `isArtifact`.

Reply:
Instead of using `isSynthetic` and then trying to get back warnings that should show up, could we do something more specific for case classes?

I'm pushing an attempt, let me know what you think.

Reply:
Yes, that is narrowly focused to address that issue. There is no reason to over-generalize.